src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -355,11 +348,26 @@ public Instance instance() throws IOException {
       if (instanceTemplate.getProperties() != null
           && instanceTemplate.getProperties().getMetadata() != null
           && instanceTemplate.getProperties().getMetadata().getItems() != null) {
+        List<Metadata.Items> instanceItems = instance.getMetadata().getItems();
         instanceTemplate
             .getProperties()
             .getMetadata()
             .getItems()
-            .forEach(instance.getMetadata().getItems()::add);
+            .forEach(
+                item -> {
+                  if (item.getKey().equals(SSH_METADATA_KEY)) {
```

Comment:
I wonder if this approach is useful for any existing items not just the SSH KEY. @ingwarsw what are your thoughts?

Reply:
Even if there is not a current additional use case, pulling it out into a helper method is a good idea for readability here. We almost do this with mergeMetadataItems in the ComputeClient, but it replaces rather than concatenating them

Reply:
moved to an helper method. at the moment only ssh keys are merged, other metadata values are overwritten